env/env.go

```diff
@@ -10,8 +10,14 @@ func IsAppEngine() bool {
 	return set
 }
 
+// Extract google cloud project from environment
+// 	 Note: this is separate from AppEngine project since other tools/services use this env var
+func GoogleCloudProject() string {
+	return os.Getenv("GOOGLE_CLOUD_PROJECT")
+}
+
 // AppEngineProject returns the cloud project GAE App is running in
 // 	 Note: this is the official way to get this information within GAE.
 func AppEngineProject() string {
-	return os.Getenv("GOOGLE_CLOUD_PROJECT")
+	return GoogleCloudProject()
```

Comment:
I don't really understand why you need a one-line function that calls another functions. Perhaps just call `GoogleCloudProject()` directly?

Reply:
This is an alias, essentially. Just `GOOGLE_CLOUD_PROJECT` set doesn't necessarily mean AppEngineEnvironment because we use the same env vars for tooling _and_ runtimes.

So I want the code to be explicit that "here we need to get the AppEngine project" vs "here we need to see if the env var is set"

Make sense?

Reply:
Sorry, I don't think I understand the difference between 'Google Cloud project' and 'App Engine project'.

This environment variable gets set by App Engine automatically and corresponds to the GCP project that the app belongs to. It is used as the default value for the following configuration options:
- Datastore project for metric records (unless `DATASTORE_PROJECT` is set)
- Stackdriver project for internal stats (unless `SD_PROJECT_FOR_INTERNAL_METRICS` is set)
- Stackdriver project for collected metrics (unless stackdriver destination has a `project_id` field set)

I see the value in adding a separate function to read the environment variable instead of calling `os.Getenv` several times, but I don't understand why we need two separate functions for this.

Reply:
So I wanted to make sure we don't mix up intentional use of it inside some tool outside of AppEngine. But now that I think about it - we haven't really had this thing run outside AppEngine so it doesn't make sense to separate this. Sorry for all the confusion on this one :|